### PR TITLE
Use \term and \defnx more correctly and consistently

### PR DESCRIPTION
**Pull request by Eelis**
No visual difference in the PDF.

### CODE REVIEW

**Comment by Eelis** [comment]
Please hold, spotted a few more

**Comment by Eelis** [comment]
Added a commit that also fixes abuse of \term for things that are not terms at all.

Only visual difference now is erasure of some inappropriate trailing whitespace resulting from the fact that in addition to making text italic, \term also includes a trailing \xspace.

**Comment by zygoloid** [comment]
Looks good other than that we would prefer the more semantic `\placeholder` be used rather than `\textit` in the incorrect uses of `\term` for placeholders.

**Comment by Eelis** [comment]
Ah, thanks for the review, will fix (and rebase).

**Comment by Eelis** [comment]
Fixed and rebased. (Still no visual difference apart from the trailing whitespace.)

**Comment by Eelis** [comment]
Bleh, missed a few, so not completely fixed yet

**Comment by Eelis** [comment]
Added fixes for a whole bunch more occurrences

File: source/atomics.tex [hunk 1]
```diff
@@ -238,71 +238,71 @@
...
 \tcode{memory_order_seq_cst}.
 \end{itemize}
 
-\begin{note} Although it is not explicitly required that \textit{S} include locks, it can
+\begin{note} Although it is not explicitly required that \placeholder{S} include locks, it can
 always be extended to an order that does include lock and unlock operations, since the
 ordering between those is already included in the ``happens before'' ordering. \end{note}
 
 \pnum
-For an atomic operation \textit{B} that reads the value of an atomic object \textit{M},
```
Comment (Eelis) [inline source/atomics.tex]: Ah no worries, I'm glad to hear there's coherent plans being formed for the use of all these things going forward. :)  I'll split out the \term stuff tonight, cheers!

**Comment by jensmaurer** [comment]
Sorry, hadn't noticed this before. #1061 and #1062 as well as #1060 might conflict.

**Comment by Eelis** [comment]
All right, I've split it up more, and now it only uses \placeholder to replace bad uses of \term.

**Comment by tkoeppe** [comment]
@jensmaurer: As the standing expert on all things `\term`, could you please review this PR?

File: source/lex.tex [hunk 2]
```diff
@@ -1726,93 +1726,93 @@
...
-operator "" @\term{X}@(@\term{f}@L)
+operator "" @\placeholder{X}@(@\placeholder{f}@L)
 \end{codeblock}
 
-Otherwise, \term{S} shall contain a raw literal operator or a literal operator
-template~(\ref{over.literal}) but not both. If \term{S} contains a raw literal operator,
-the \term{literal} \term{L} is treated as a call of the form
+Otherwise, \placeholder{S} shall contain a raw literal operator or a literal operator
+template~(\ref{over.literal}) but not both. If \placeholder{S} contains a raw literal operator,
+the \placeholder{literal} \placeholder{L} is treated as a call of the form
```
Comment (jensmaurer) [inline source/lex.tex]: \term{literal} should be \grammarterm{literal}, not \placeholder

File: source/lib-intro.tex [hunk 3]
```diff
@@ -604,20 +604,20 @@
...
 
-static const @\term{enumerated C0}@ (@\term{V0}@);
-static const @\term{enumerated C1}@ (@\term{V1}@);
-static const @\term{enumerated C2}@ (@\term{V2}@);
-static const @\term{enumerated C3}@ (@\term{V3}@);
+static const @\placeholder{enumerated C0}@ (@\placeholder{V0}@);
+static const @\placeholder{enumerated C1}@ (@\placeholder{V1}@);
+static const @\placeholder{enumerated C2}@ (@\placeholder{V2}@);
+static const @\placeholder{enumerated C3}@ (@\placeholder{V3}@);
   .....
```
Comment (jensmaurer) [inline source/lib-intro.tex]: Those are two distinct placeholders. Please try separating them (i.e. have two \placeholder's next to each other). Does that impair the spacing?

File: source/lib-intro.tex [hunk 4]
```diff
@@ -637,56 +637,56 @@
...
 
-constexpr @\term{bitmask C0}@(@\term{V0}{}@);
-constexpr @\term{bitmask C1}@(@\term{V1}{}@);
-constexpr @\term{bitmask C2}@(@\term{V2}{}@);
-constexpr @\term{bitmask C3}@(@\term{V3}{}@);
+constexpr @\placeholder{bitmask C0}@(@\placeholder{V0}{}@);
+constexpr @\placeholder{bitmask C1}@(@\placeholder{V1}{}@);
+constexpr @\placeholder{bitmask C2}@(@\placeholder{V2}{}@);
+constexpr @\placeholder{bitmask C3}@(@\placeholder{V3}{}@);
   .....
```
Comment (jensmaurer) [inline source/lib-intro.tex]: Same here: two \placeholders, once for each meta-variable.

File: source/regex.tex [hunk 5]
```diff
@@ -3916,7 +3916,7 @@
 sequence \end{note}
 
 \pnum
-The \textit{current match} is \tcode{(*position).prefix()} if \tcode{subs[N] == -1}, or
+The \term{current match} is \tcode{(*position).prefix()} if \tcode{subs[N] == -1}, or
 \tcode{(*position)[subs[N]]} for any other value of \tcode{subs[N]}.
 
 \rSec3[re.tokiter.cnstr]{\tcode{regex_token_iterator} constructors}
```
Comment (jensmaurer) [inline source/regex.tex]: This patch adds \term, which this issue tries to eradicate (admittedly morphed from even worse \textit). Please drop this patch from here for now; I expect those \textit to mostly become \defn or \defnx.

File: source/classes.tex [hunk 6]
```diff
@@ -228,14 +225,13 @@
...
 standard-layout class, and has no non-static data members of type non-POD struct,
 non-POD union (or array of such types). Similarly, a
-\term{POD union} is a union that is both a trivial class and a standard-layout
+\defn{POD union} is a union that is both a trivial class and a standard-layout
 class, and has no non-static data members of type non-POD struct, non-POD
-union (or array of such types). A \term{POD class} is a
+union (or array of such types). A \defn{POD class} is a
 class that is either a POD struct or a POD union.
 
 \begin{example}
```
Comment (jensmaurer) [inline source/classes.tex]: Do we have any rules when to use "POD~struct" vs. "POD struct" in the index? We seem to prefer the variant with the tilde = non-breaking space.  If true, the stuff above needs to become \defnx.

File: source/lex.tex [hunk 7]
```diff
@@ -870,26 +870,21 @@
...
+An \defnx{octal}{literal!octal} integer
 literal (base eight) begins with the digit \tcode{0} and consists of a
 sequence of octal digits.\footnote{The digits \tcode{8} and \tcode{9} are not octal digits. }
-A \term{decimal}
+A \defnx{decimal}{literal!decimal}
 integer literal (base ten) begins with a digit other than \tcode{0} and
 consists of a sequence of decimal digits.
-A \term{hexadecimal} integer literal (base sixteen) begins with
+A \defnx{hexadecimal}{literal!hexadecimal} integer literal (base sixteen) begins with
 \tcode{0x} or \tcode{0X} and consists of a sequence of hexadecimal
```
Comment (jensmaurer) [inline source/lex.tex]: Please put all of "XXX integer literal" into the first argument of \defnx; we're not actually defining the term "decimal" here, but the phrase "decimal integer literal".

File: source/regex.tex (hunk 5 above)
Comment (Eelis) [inline source/regex.tex]: Ah ok, I didn't know there was a plan to eradicate \term.

File: source/utilities.tex [hunk 8]
```diff
@@ -12224,30 +12224,24 @@
 The following definitions apply to this Clause:
 
 \pnum
-\indexdefn{call signature}%
-A \term{call signature} is the name of a return type followed by a
+A \defn{call signature} is the name of a return type followed by a
 parenthesized comma-separated list of zero or more argument types.
```
Comment (jensmaurer) [inline source/utilities.tex]: For this (and those that follow), the index entries should probably use tilde instead of space.

**Comment by jensmaurer** [comment]
Review complete; see comments above. Going to bed now.

File: source/classes.tex (hunk 6 above)
Comment (Eelis) [inline source/classes.tex]: (If that's an actual goal, though, there are hundreds more occurrences that would need to be replaced.)

**Comment by Eelis** [comment]
Hm, this pull request has really lost focus. I originally just wanted to use \defn* for \terms that already had index entries, a change which I could make without affecting those index entries or the formatting of the text, but I don't feel qualified to make the kind of changes to the index entries themselves (by changing their use of tildes) or to the formatting of the text (by changing which parts of the text are italicized) that Jens is requesting, so probably better to just go with his patches. :)

**Pull request closed without merging**